tests/test_api.rs

```diff
@@ -1949,6 +1949,18 @@ fn fn_callback(
   rv.set(s.into());
 }
 
+fn fn_callback_new(
+  scope: &mut v8::HandleScope,
+  args: v8::FunctionCallbackArguments,
+  mut rv: v8::ReturnValue,
+) {
+  assert_eq!(args.length(), 0);
+  assert!(!args.new_target().is_undefined());
```

Comment:
Assert that it is an object.
I'd also try to set a property on the object, and then assert that the object returned by `new f2()` has that property.